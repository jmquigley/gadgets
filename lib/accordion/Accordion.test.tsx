'use strict';

import {shallow} from 'enzyme';
import * as React from 'react';
import {Accordion, getDefaultAccordionProps} from '../../dist/bundle';

test('Test retrieval of Accordion props object', () => {
	const props = getDefaultAccordionProps();

	expect(props).toBeDefined();
	expect(props).toMatchSnapshot();
});

test('Test the creation of a Accordion control container', () => {
	const li = <li>some list item</li>;
	const ctl = shallow(
		<Accordion id="1234" className="test-class">
			{li}
		</Accordion>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
});

test('Test the disabling of the Accordion control container', () => {
	const li = <li>some list item</li>;
	const ctl = shallow(
		<Accordion
			className="test-class"
			disabled
			id="1234"
		>
			{li}
		</Accordion>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
});

test('Test making the accordion control invisible', () => {
	const li = <li>some list item</li>;
	const ctl = shallow(
		<Accordion
			className="test-class"
			id="1234"
			visible={false}
		>
			{li}
		</Accordion>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
});
