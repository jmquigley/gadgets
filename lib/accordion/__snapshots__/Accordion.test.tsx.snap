// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test making the accordion control invisible 1`] = `
<Wrapper
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className="test-class"
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  id="1234"
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Accordion"
  onError={[Function]}
  padding={null}
  reset={false}
  ripple={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={false}
  width={null}
  xcss={null}
>
  <Accordion__AccordionView
    className="ui-accordion test-class"
    sizing="normal"
    style={Object {}}
  >
    <li
      disabled={false}
      sizing="normal"
      visible={false}
    >
      some list item
    </li>
  </Accordion__AccordionView>
</Wrapper>
`;

exports[`Test retrieval of Accordion props object 1`] = `
Object {
  "backgroundColor": null,
  "border": null,
  "borderColor": null,
  "bottom": null,
  "children": null,
  "className": null,
  "color": null,
  "controlled": true,
  "direction": "right",
  "disabled": false,
  "err": null,
  "errorMessage": null,
  "focus": false,
  "height": null,
  "hidden": false,
  "id": null,
  "left": null,
  "location": "none",
  "maxHeight": null,
  "maxWidth": null,
  "minHeight": null,
  "minWidth": null,
  "noborder": false,
  "noedit": false,
  "nohover": false,
  "nopropagation": false,
  "notheme": false,
  "notooltip": false,
  "obj": "Accordion",
  "padding": null,
  "ripple": false,
  "selected": false,
  "sizing": "normal",
  "style": Object {},
  "testing": true,
  "theme": Object {
    "backgroundColor": "#ffffff",
    "borderColor": "#c0c0c0",
    "buttonColor": "#666666",
    "chevronColor": "#cccccc",
    "color": "#000000",
    "fontFamily": "OpenSans, Georgia, serif",
    "headerBackgroundColor": "#004358",
    "headerForegroundColor": "#ffffff",
    "headerHoverColor": "#ffe11a",
    "hoverColor": "#c0c0c0",
    "inputBorderColor": "#c0c0c0",
    "itemHoverColor": "#bedb39",
    "outlineColor": "#51cbee",
    "searchFocus": "#51cbee",
    "searchMatch": "#fd7400",
    "selectedBackgroundColor": "#1f8a70",
    "selectedForegroundColor": "#ffe11a",
    "titleBarBackgroundColor": "#004358",
    "titleBarForegroundColor": "#ffe11a",
    "tooltipBackgroundColor": "#708090",
    "tooltipForegroundColor": "#ffffff",
    "transitionDelay": "0.33s",
  },
  "tooltip": null,
  "top": null,
  "visible": true,
  "width": null,
  "xcss": null,
}
`;

exports[`Test retrieval of Accordion props object 2`] = `
Object {
  "children": null,
  "sizing": "normal",
  "style": Object {},
}
`;

exports[`Test the creation of a Accordion control container 1`] = `
<Wrapper
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className="test-class"
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  id="1234"
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Accordion"
  onError={[Function]}
  padding={null}
  reset={false}
  ripple={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <Accordion__AccordionView
    className="ui-accordion test-class"
    sizing="normal"
    style={Object {}}
  >
    <li
      disabled={false}
      sizing="normal"
      visible={true}
    >
      some list item
    </li>
  </Accordion__AccordionView>
</Wrapper>
`;

exports[`Test the disabling of the Accordion control container 1`] = `
<Wrapper
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className="test-class"
  color={null}
  controlled={true}
  direction="right"
  disabled={true}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  id="1234"
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Accordion"
  onError={[Function]}
  padding={null}
  reset={false}
  ripple={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <Accordion__AccordionView
    className="ui-accordion test-class nohover"
    sizing="normal"
    style={Object {}}
  >
    <li
      disabled={true}
      sizing="normal"
      visible={true}
    >
      some list item
    </li>
  </Accordion__AccordionView>
</Wrapper>
`;
